modules/gapi/include/opencv2/gapi/python_bridge.hpp

```diff
@@ -0,0 +1,233 @@
+                                            , float
+                                            , std::string
+                                            , cv::Point
+                                            , cv::Point2f
+                                            , cv::Size
+                                            , cv::Rect>;
+    template<typename T>
+    GOpaqueT(cv::GOpaque<T> arg) : m_arg(arg) { };
+
+    GAPI_WRAP GOpaqueT(gapi::ArgType type) : m_type(type)
+    {
+
+#define HANDLE_CASE(T, K) case gapi::ArgType::CV_##T: \
+                m_arg = cv::GOpaque<K>(); \
+                break;
+
+        switch (type)
+        {
+            HANDLE_CASE(BOOL,    bool);
+            HANDLE_CASE(INT,     int);
+            HANDLE_CASE(DOUBLE,  double);
+            HANDLE_CASE(FLOAT,   float);
+            HANDLE_CASE(STRING,  std::string);
+            HANDLE_CASE(POINT,   cv::Point);
+            HANDLE_CASE(POINT2F, cv::Point2f);
+            HANDLE_CASE(SIZE,    cv::Size);
+            HANDLE_CASE(RECT,    cv::Rect);
+#undef HANDLE_CASE
+            default:
+                GAPI_Assert(false && "Unsupported type");
+        }
+    }
+
+    cv::detail::GOpaqueU strip() {
+#define HANDLE_CASE(T) case Storage::template index_of<cv::GOpaque<T>>(): \
+        return cv::util::get<cv::GOpaque<T>>(m_arg).strip(); \
+
+        switch (m_arg.index())
+        {
+            HANDLE_CASE(bool);
+            HANDLE_CASE(int);
+            HANDLE_CASE(double);
+            HANDLE_CASE(float);
+            HANDLE_CASE(std::string);
+            HANDLE_CASE(cv::Point);
+            HANDLE_CASE(cv::Point2f);
+            HANDLE_CASE(cv::Size);
+            HANDLE_CASE(cv::Rect);
+#undef HANDLE_CASE
+            default:
+                GAPI_Assert(false && "Unsupported type");
+        }
+        GAPI_Assert(false);
+    }
+
+GAPI_WRAP gapi::ArgType type() { return m_type; }
+
+private:
+    gapi::ArgType m_type;
+    Storage m_arg;
+};
+
+class GAPI_EXPORTS_W_SIMPLE GArrayT
+{
+public:
+    using Storage = detail::MakeVariantType<  cv::GArray
+                                            , bool
+                                            , int
+                                            , double
+                                            , float
+                                            , std::string
+                                            , cv::Point
+                                            , cv::Point2f
+                                            , cv::Size
+                                            , cv::Rect
+                                            , cv::Scalar
+                                            , cv::Mat
+                                            , cv::GMat>;
+    template<typename T>
+    GArrayT(cv::GArray<T> arg) : m_arg(arg) { };
+
+    GAPI_WRAP GArrayT(gapi::ArgType type) : m_type(type)
+    {
+
+#define HANDLE_CASE(T, K) case gapi::ArgType::CV_##T: \
+                m_arg = cv::GArray<K>(); \
```

Comment:
Can some of this code become template to support both array and opaque? All those duplicates of `HANDLE_CASE` with the same types looks way too massive

Reply:
E.g. move some switches to template functions which return desired type and then use those functions in constructors/etc

Reply:
What should return this template function ?

Reply:
Can try to utilize something like `getFromStream`, non-blocking though